src/js/actions/tools.js

```diff
@@ -446,10 +464,16 @@ define(function (require, exports) {
     };
     selectTool.reads = [];
     selectTool.writes = [locks.JS_TOOL, locks.PS_TOOL];
-    selectTool.transfers = [installShapeDefaults, shortcuts.addShortcut,
-        shortcuts.removeShortcut, "layers.deleteSelected", "layers.resetLayers",
-        policy.removePointerPolicies, policy.removeKeyboardPolicies, policy.addPointerPolicies,
-        policy.addKeyboardPolicies, policy.setMode];
+    selectTool.transfers = [swapPolicies, policy.removePointerPolicies,
```

Comment:
It would be great if there were a way to say "all the tools? the select/deselect actions on those". The only downside is that the list of tools' actions has to be specified up front. So, I think there's definitely room for improvement, but I think it is clearly preferable to what is exists on master today.